src/test/kotlin/com/atlassian/performance/tools/lib/Apdex.kt

```diff
@@ -12,17 +12,30 @@ class Apdex {
     fun score(
         metrics: List<ActionMetric>
     ): Double {
-        return metrics
-            .filter { it.result == ActionResult.OK }
-            .map { score(it) }
+        return average(weight(metrics
+            .filter { it.result == ActionResult.OK }))
+    }
+
+    fun average(
+        weightedMetrics: List<WeightedActionMetric>
+    ): Double {
+        return weightedMetrics
+            .map { it.weight }
             .average()
     }
 
+    fun weight(
```

Comment:
In general, functions should be named after activities (verb-like), so we'd go for `weigh`

Reply:
done 👍